lib/kubeclient/watch_stream.rb

```diff
@@ -19,12 +19,13 @@ def each
 
         @http.request(request) do |response|
           unless response.is_a? Net::HTTPSuccess
-            fail KubeException.new(response.code, response.message)
+            fail KubeException.new(response.code, response.message, response)
           end
+          is_json = response['Content-Type'] == 'application/json'
```

Comment:
Arg, I just discovered this doesn't necessarily work with all watch endpoints.  Coming up with a new solution...